Yoast/YoastSEO.js
# Improve keyword recognition in Turkish

## Summary

This PR can be summarized in the following changelog entry:

* Improves keyword recognition in Turkish by providing a customized transliteration map for Turkish characters `İ` / `i` and `I` / `ı`. These characters are not supported by standard regexes. 

## Relevant technical choices:

* Generates a list of all possible usages of the special characters, including case manipulation (e.g., by keyword "İstanbul" the word "istanbul" will still be found in the text and vice versa) and latinization (e.g., by keyword "İstanbul" the word "Istanbul" will still be found in the text).

## Test instructions

This PR can be tested by following these steps:

* In Browserified and in the beta.
* Take a text in Turkish and use a word that starts with `İ`, `i`, `I` and `i`  as your keyword. Don't forget to specify `tr_TR` as your locale.
* Try manipulating the case and latinization in the text and check if the keyword recognition works well.
* Check if the markings work correctly.
* Check if this new functionality does not influence keyword recognition for other locales.

Fixes #1193 Fixes https://github.com/Yoast/wordpress-seo/issues/10384
